src/python/WMCore/WMSpec/StdSpecs/StepChain.py

```diff
@@ -321,9 +324,17 @@ def setupNextSteps(self, task, origArgs):
                 multicore = taskConf['Multicore']
             if taskConf.get("EventStreams") is not None and taskConf['EventStreams'] >= 0:
                 eventStreams = taskConf['EventStreams']
-
             currentCmsswStepHelper.setNumberOfCores(multicore, eventStreams)
 
+            # GPU settings
+            gpuRequired = self.requiresGPU
+            gpuParams = json.loads(taskConf.get('GPUParams', None))
```

Comment:
This is dangerous. It may raise the following exceptions:
* `TypeError: the JSON object must be str, bytes or bytearray, not NoneType` - if the `GPUParams` is missing
* `JSONDecodeError: Expecting value: line 1 column 1 (char 0)` - if the key `GPUParams` is present but is an empty or non json formatted string

If we have those cases covered in some earlier checks that's ok, but I'd stay on the safe side and would do something like:
```
if taskConf.get('GPUParams', None):
    gpuParams = json.loads(taskConf['GPUParams'])
else:
    gpuParams = json.loads(self.gPUParams)
```
which would also cover the check from two lines bellow.

Reply:
Good catch Todor! Let me set it to the json encoded representation of None, which is `'null'`.